src/main/java/CLI/Main.java

```diff
@@ -375,4 +387,91 @@ private static void checkHistoryAddJobsAndDownload(Job job, boolean removeInputB
             downloader.run();
         }
     }
-}
+
+    public static boolean checkUpdate() throws URISyntaxException {
+        String latestVersion = getLatestVersion();
+        if (isNewerVersion(latestVersion , VERSION_NUMBER)) {
+            String Link;
+            String oldFilePath = String.valueOf(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+            String newFilePath = "";
+            String OS_NAME = OS.getOSName();
+            if (OS_NAME.contains("win")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI.exe";
+                String fileName = "Drifty_CLI.exe";
+                String dirPath = String.valueOf(Paths.get(System.getenv("LOCALAPPDATA"), "Drifty", "updates"));
+                FileDownloader downloader =  new FileDownloader(Link , fileName , dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            } else if (OS_NAME.contains("mac")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_macos";
+                String fileName = "Drifty-CLI_macos";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader =  new FileDownloader(Link , fileName , dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            } else if (OS_NAME.contains("linux")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_linux";
+                String fileName = "Drifty-CLI_linux";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader = new FileDownloader(Link, fileName, dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            }
```

Comment:
@kmr-rohit Please extract the common parts from these three if-else blocks. It helps in reducing redundancy.
- Consider adding an array with the name of the executables for each OS and retrieving the array elements as needed.
- The FileDownloader statements can be extracted.
- The `dirPath` will be different for different OS. 
     - For Windows, it will be `Paths.get(System.getenv("LOCALAPPDATA"), "Drifty", "updates")`
     - For Linux/macOS, it will be `.drifty/updates`
- The `dirPath` will remain the same in all three if-else statements. So, it can be extracted and kept before the starting of the if-else blocks.

Reply:
Tried to eliminate redundancy , written more optimized code

Reply:
Okay :+1: